src/coord/src/coord.rs

```diff
@@ -289,21 +295,46 @@ where
         Ok(coord)
     }
 
-    pub fn now(&mut self) -> Timestamp {
-        let mut t = self
+    /// Assign a timestamp for a read.
+    fn get_read_ts(&mut self) -> Timestamp {
+        let ts = self.get_ts();
+        self.last_op_was_read = true;
+        self.read_lower_bound = ts;
+        ts
+    }
+
+    /// Assign a timestamp for a write. Writes following reads must ensure that they are assigned a
+    /// strictly larger timestamp to ensure they are not visible to any real-time earlier reads.
+    fn get_write_ts(&mut self) -> Timestamp {
+        if self.last_op_was_read {
+            self.last_op_was_read = false;
+            let ts = self.get_ts();
+            if ts <= self.read_lower_bound {
+                self.read_lower_bound + 1
+            } else {
+                self.read_lower_bound = ts;
```

Comment:
can you help me understand why we need to assign to `read_lower_bound` on line 315, but not on line 313? like if the read lower bound was 7, and a write makes its way to line 313 and writes at ts = 8, it seems like a subsequent read could still read from ts = 7 because we didn't update `read_lower_bound`? what am i missing?

Reply:
Yeah I think you're right, in cleaning this up I have changed it a bit and I think it's fine now, we always update `read_lower_bound`.